statics/workflows/ping-mesh.yaml

```diff
@@ -0,0 +1,103 @@
+      - description: "Protocol : TCP/IPv4"
+        value: TCP
+      - description: "Protocol : UDP/IPv4"
+        value: UDP	
+source: |
+    function PingMesh(from, to, protocol) {
+      var dict = {},
+            From = {},
+            sources = [],
+            captureuuids = [],
+            i = 1,
+            randvalue = Math.floor(Math.random() * 4096);
+      var capture = new Capture();
+      var packetInjection = new PacketInjection();
+      sources.push(from);
+      sources.forEach(function(source){
+        capture.GremlinQuery = "G.V().Has('TID', '" + source + "').ShortestPathTo(Metadata('TID', '" + to + "'))";
+        packetInjection.Src = "G.V().Has('TID', '" + source + "')"
+        packetInjection.Dst = "G.V().Has('TID', '" + to + "')"
+        packetInjection.Count = 5
+        return client.captures.create(capture).then(function (c) {
+            capture = c
+            captureuuids.push(capture.UUID);
+            return sleep(2000)
+        }).then(function () {
+            return client.G.V().Has("TID", source)
+        }).then(function (nodes) {
+            if (nodes[0].Metadata.Neutron && nodes[0].Metadata.Neutron.IPV4) {
+                packetInjection.SrcIP = nodes[0].Metadata.Neutron.IPV4[0]
+            }
+            if (nodes[0].Metadata.ExtID && nodes[0].Metadata.ExtID["attached-mac"]) {
+                packetInjection.SrcMAC = nodes[0].Metadata.ExtID["attached-mac"]
+            }
+            if (protocol == "ICMPv4") {
+                packetInjection.Type = "icmp4"
+                packetInjection.ICMPID = randvalue + i;
+                From[source] = packetInjection.ICMPID;
+            }
+            if (protocol == "TCP") {
+                packetInjection.Type = "tcp4"
+                packetInjection.SrcPort = 16384+randvalue + i;
+                packetInjection.DstPort = 32768+randvalue + i;
+                From[source] = packetInjection.DstPort;
+            } 
+            if (protocol == "UDP") {
+                packetInjection.Type = "udp4"
+                packetInjection.SrcPort = 16384+randvalue;
+                packetInjection.DstPort = 32768+randvalue;
+                From[source] = packetInjection.DstPort;
+            }
+            return client.G.V().Has("TID", to)
+        }).then(function (nodes) {
+            if (nodes[0].Metadata.Neutron && nodes[0].Metadata.Neutron.IPV4) {
+                packetInjection.DstIP = nodes[0].Metadata.Neutron.IPV4[0]
+            }
+            if (nodes[0].Metadata.ExtID && nodes[0].Metadata.ExtID["attached-mac"]) {
+                packetInjection.DstMAC = nodes[0].Metadata.ExtID["attached-mac"]
+            }
+            return client.packetInjections.create(packetInjection)
+        })
+        i += 1;
+      });
+      return sleep(2000).then(function () {
+       sources.forEach(function(source){
+        var flows = [];
+        if (protocol == "ICMPv4") {
+            flows.push(client.G.Flows().Has("ICMP.ID", From[source]));
```

Comment:
you don't need flows hash

var flows = client.G.Flows().Has("ICMP.ID", From[source])
flows.length > 0